$MirOS: ports/devel/glib/patches/patch-glib_gslice_c,v 1.2 2009/07/07 18:08:02 bsiegert Exp $
--- glib/gslice.c.orig	Fri Sep 12 19:55:28 2008
+++ glib/gslice.c	Thu Nov 25 12:24:44 2010
@@ -103,6 +103,7 @@
 /* --- macros and constants --- */
 #define LARGEALIGNMENT          (256)
 #define P2ALIGNMENT             (2 * sizeof (gsize))                            /* fits 2 pointers (assumed to be 2 * GLIB_SIZEOF_SIZE_T below) */
+#undef ALIGN /* from <machine/param.h> */
 #define ALIGN(size, base)       ((base) * (gsize) (((size) + (base) - 1) / (base)))
 #define NATIVE_MALLOC_PADDING   P2ALIGNMENT                                     /* per-page padding left for native malloc(3) see [1] */
 #define SLAB_INFO_SIZE          P2ALIGN (sizeof (SlabInfo) + NATIVE_MALLOC_PADDING)
@@ -1110,8 +1111,10 @@ slab_allocator_free_chunk (gsize    chun
 }
 
 /* --- memalign implementation --- */
+#if HAVE_MEMALIGN
 #ifdef HAVE_MALLOC_H
 #include <malloc.h>             /* memalign() */
+#endif
 #endif
 
 /* from config.h:
